app/domain/authentication/authn_jenkins/authenticator.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+require 'json'
+
+module Authentication
+  module AuthnJenkins
+    class Authenticator
+
+      Err = Errors::Authentication::AuthnJenkins
+      def initialize(env:)
+        @env = env
+      end
+
+      def webservice
+        @webservice ||= ::Authentication::Webservice.new(
+          account:            @account,
+          authenticator_name: @authenticator_name,
+          service_id:         @service_id
+        )
+      end
+
+      def http_get_request(url, username, password) 
```

Comment:
Let's try to get all the functionality related Jenkins out into it's own class.  This accomplishes a couple of things:
- Provides better abstraction
- Simplifies authenticator tests as you can mock the requests to Jenkins

Something like:
```ruby
class JenkinsClient
  def initialize(url, username, password)
    @url = url
    @username = username
    @password = password
  end

  def public_key()
    response = request('')
    create_public_key(response['X-Instance-Identity'])
  end

  def build(job, build_number)
    request("/job/#{job}/#{build_number}/api/json")
  end

  private
  def create_public_key(identity)
    public_key = [
      "-----BEGIN PUBLIC KEY-----",
      identity,
      "-----END PUBLIC KEY-----"
    ].join("\n")
    Rails.logger.debug("Returned public key: #{public_key}")
    OpenSSL::PKey::RSA.new(public_key)
  end

  def request(path)
    # Uses the HTTP gem: https://github.com/httprb/http
    begin
      HTTP.basic_auth(user: @username, pass: @password).get("#{@url}/#{path}")
    rescue Net::OpenTimeout => e
      Rails.logger.error("Timeout to Jenkins host Exception #{e}. Jenkins host '#{url}' is most likely invalid. Validate this url can be accessed from the conjur server.")
      raise Err::HostNotFound, url
    end
  end
end
```

Reply:
Added jenkins client in this commit: https://github.com/AndrewCopeland/conjur/commit/d16c40c290d4f1ac139b101cd089a1635f20db29